buildkite/bazelci.py

```diff
@@ -766,17 +766,31 @@ def execute_commands(
             raise BuildkiteException("There are neither build nor test targets")
 
         if build_targets:
-            execute_bazel_build(
-                bazel_version,
-                bazel_binary,
-                platform,
-                task_config.get("build_flags", []),
-                build_targets,
-                None,
-                incompatible_flags,
-            )
-            if save_but:
-                upload_bazel_binary(platform)
+            json_profile_flags = []
+            include_json_profile = task_config.get("include_json_profile", False)
+
+            if include_json_profile:
+                json_profile_temp_out = os.path.join(tmpdir, "build.profile")
+                json_profile_flags = [
+                    "--experimental_generate_json_trace_profile",
+                    "--experimental_profile_cpu_usage",
+                    "--profile={}".format(json_profile_temp_out)
```

Comment:
you probably should enable compression

Reply:
and test out how big the profile is for a big project to see if you want to enable slim profiling

Reply:
for bazelbuild/bazel, the json profile for building `//src:bazel` and `//src:bazel_jvm_minimal` was just 397KB

Reply:
How much is it for this?

```
bazel test --flaky_test_attempts=3 --build_tests_only  --keep_going --experimental_multi_threaded_digest --experimental_repository_cache_hardlinks --sandbox_tmpfs_path=/tmp --experimental_build_event_json_file_path_conversion=false --build_event_json_file=/tmp/tmpkot74tw9/test_bep.json --test_timeout=1200 --test_env=HOME -- //scripts/... //src/java_tools/... //src/test/... //src/tools/singlejar/... //third_party/ijar/... //tools/android/... //tools/aquery_differ/... //tools/python/... -//src/test/shell/bazel:embedded_tools_deps_test -//src/java_tools/buildjar/... -//src/java_tools/import_deps_checker/...
```

In your change, we're not creating/uploading the profile for tests, are we? Do you want to do this separately?

Reply:
New changes:
- Compressing results by default
- Added support for test